src/lib/hooks/useCurrency.ts

```diff
@@ -105,8 +105,8 @@ export function useCurrencyFromMap(tokens: TokenMap, currencyId?: string | null)
 
   const token = useTokenFromMapOrNetwork(tokens, isNative ? undefined : shorthandMatchAddress ?? currencyId)
 
-  const chainAllowed = chainId && isChainAllowed(chainId)
-  if (currencyId === null || currencyId === undefined || !chainAllowed) return null
+  const supportedChain = isSupportedChain(chainId)
+  if (currencyId === null || currencyId === undefined || !supportedChain) return null
```

Comment:
non blocking: we could move this check after the useWeb3React hook to potentially return null before setting isNative, token, etc.

Reply:
we can't because all the hooks need to happen without early returns in react unfortunately

Reply:
Sorry token would have to be before but it seems like `shorthandMatchAddress` and `isNative` both don't use hooks

Reply:
See their reqs in token

Reply:
`shorthandMatchAddress` uses `useMemo` though. `isNative` is used in the `useTokenFromMapOrNetwork` call so it also can't get moved too much.